etl/src/main/java/org/pih/hivmigration/etl/sql/ExamLabResultsMigrator.groovy

```diff
@@ -142,6 +142,60 @@ class ExamLabResultsMigrator extends ObsMigrator {
         ''')
 
         migrate_tmp_obs()
+        
+        //
+        // -------- RPR
+        //
+
+        create_tmp_obs_table()
+
+        executeMysql("Migrate coded rpr", '''
+            INSERT INTO tmp_obs (source_encounter_id, concept_uuid, value_coded_uuid)
+            SELECT
+                he.source_encounter_id,
+                concept_uuid_from_mapping('CIEL', '1619'),
+                CASE result
+                    WHEN 'negative' THEN concept_uuid_from_mapping('PIH', 'NON-REACTIVE')
+                    WHEN 'positive' THEN concept_uuid_from_mapping('PIH', 'REACTIVE')
+                    END
+            FROM hivmigration_exam_lab_results helr
+            JOIN hivmigration_encounters he on helr.source_encounter_id = he.source_encounter_id
+            JOIN encounter e on he.encounter_id = e.encounter_id
+            WHERE
+                lab_test = 'rpr'
+                AND result IN ('negative', 'positive');
+        ''')
+
+        executeMysql("Log warning about invalid rpr values", '''
+            INSERT INTO hivmigration_data_warnings (openmrs_patient_id, openmrs_encounter_id, field_name, field_value, warning_type)
+            SELECT
+                hp.person_id,
+                he.encounter_id,
+                'HIV_EXAM_LAB_RESULTS rpr',
+                helr.result,
+                'Invalid value. Not migrated.'
+            FROM hivmigration_exam_lab_results helr
+            JOIN hivmigration_encounters he on helr.source_encounter_id = he.source_encounter_id
+            JOIN hivmigration_patients hp on he.source_patient_id = hp.source_patient_id
+            WHERE lab_test = 'rpr' AND result NOT IN ('negative', 'positive');
+        ''')
+
+        executeMysql("Log warning about missing encounters", '''
+            INSERT INTO hivmigration_data_warnings (openmrs_patient_id, openmrs_encounter_id, field_name, field_value, warning_type)
+            SELECT
+                hp.person_id,
+                he.encounter_id,
+                'HIV_EXAM_LAB_RESULTS rpr',
+                helr.result,
+                'Belongs to an encounter that was not migrated. Value not migrated.'
+            FROM hivmigration_exam_lab_results helr
+            JOIN hivmigration_encounters he on helr.source_encounter_id = he.source_encounter_id
+            JOIN hivmigration_patients hp on he.source_patient_id = hp.source_patient_id
+            LEFT JOIN encounter e on he.encounter_id = e.encounter_id
+            WHERE lab_test = 'rpr' AND e.encounter_id IS NULL;
```

Comment:
Following on from my comment on the previous commit... we may want to have a way to not have to write the query for every different type of lab test, and just check that every encounter in the hivmigration_encounters ends up in the encounter table... especially now if we are writing a query like this for every lab test.

Will be interesting to see what this table shows... tangentially, I think these will likely be more migration errors than DQ errors... ie things we are missing in our migration and need to fix on the migration side

Reply:
https://github.com/PIH/hivmigration/pull/70